packages/core/src/external/aws/lambda-logic/modify-resources.ts

```diff
@@ -0,0 +1,599 @@
+  };
+}
+
+export type SlimLocation = Omit<Location, "address" | "type"> & {
+  address?: string | undefined;
+  type?: string | undefined;
+  reference?: Record<string, string>;
+};
+
+function slimLocation(res: Location): SlimLocation {
+  const updRes = cloneDeep(res);
+  return {
+    ...updRes,
+    address: getAddressString(res.address),
+    type: getUniqueDisplaysString(res.type),
+  };
+}
+
+function cleanUpNote(note: string): string {
+  return note
+    .trim()
+    .replace(new RegExp(REMOVE_FROM_NOTE.join("|"), "g"), "")
+    .replace(/<ID>.*?<\/ID>/g, "")
+    .replace(/<styleCode>.*?<\/styleCode>/g, "")
+    .replace(/\s*\n\s*/g, "\n")
+    .replace(/\*{2,}/g, "*")
+    .replace(/_{2,}/g, " ");
+}
+
+function getQuantityString(quantity: Quantity | undefined): string | undefined {
+  if (!quantity) return undefined;
+  return `${quantity.value}${quantity.unit ? ` ${quantity.unit}` : ""}`;
+}
+
+function getAddressString(address: Address | Address[] | undefined): string | undefined {
+  if (!address) return undefined;
+
+  return toArray(address)
+    .map(addr => `${addr.line}, ${addr.city}, ${addr.state}`)
+    .join("\n");
+}
+
+function getUniqueDisplaysString(
+  concept: CodeableConcept | CodeableConcept[] | undefined
+): string | undefined {
+  return getUniqueDisplays(concept)?.join(", ");
+}
+
+function getLongestDisplay(
+  concept: CodeableConcept | CodeableConcept[] | undefined
+): string | undefined {
+  if (!concept) return undefined;
+
+  const uniqueDescriptors = new Set<string>();
+  const concepts = toArray(concept);
+  concepts.forEach(concept => {
+    const text = concept.text;
+    if (text) uniqueDescriptors.add(text.trim().toLowerCase());
+
+    concept.coding?.forEach(coding => {
+      if (coding.display) uniqueDescriptors.add(coding.display.trim().toLowerCase());
+    });
+  });
+
+  if (uniqueDescriptors.size === 0) return undefined;
+  return Array.from(uniqueDescriptors).reduce((longest, current) =>
+    current.length > longest.length ? current : longest
+  );
+}
+
+function getUniqueDisplays(
+  concept: CodeableConcept | CodeableConcept[] | undefined
+): string[] | undefined {
+  if (!concept) return undefined;
+
+  const uniqueDescriptors = new Set<string>();
+  const concepts = toArray(concept);
+  concepts.forEach(concept => {
+    const text = concept.text;
+    if (text) uniqueDescriptors.add(text.trim().toLowerCase());
+
+    concept.coding?.forEach(coding => {
+      if (coding.display) uniqueDescriptors.add(coding.display.trim().toLowerCase());
+    });
+  });
+
+  if (uniqueDescriptors.size === 0) return undefined;
+  return Array.from(uniqueDescriptors);
+}
+
+function isUselessStatus(status: string | undefined): boolean {
+  return !status || status === "" || status === "final";
+}
+
+export function getNameString(names: HumanName | HumanName[] | undefined): string | undefined {
+  const nameParts = new Set<string>();
+  toArray(names).forEach(name => {
+    delete name.use;
```

Comment:
I think we shouldn't update the params to these functions. This is dangerous and goes against immutability - [reference](https://www.notion.so/metriport/Development-Best-Practices-bb3574d3c83d46dc9914a9e5c5542ca7?pvs=4#10eb51bb90408024bb27fe74a334c78f).

Reply:
good catch! fixed!